controller/appcontroller.go

```diff
@@ -672,11 +739,10 @@ func (ctrl *ApplicationController) processAppRefreshQueueItem() (processNext boo
 		}
 	}
 
-	conditions, hasErrors := ctrl.refreshAppConditions(app)
-	if hasErrors {
+	ctrl.refreshAppConditions(app)
+	if len(app.Status.GetErrorConditions()) > 0 {
 		app.Status.Sync.Status = appv1.SyncStatusCodeUnknown
 		app.Status.Health.Status = appv1.HealthStatusUnknown
-		app.Status.Conditions = conditions
```

Comment:
did you meant to delete this line?

Reply:
yes, the  `refreshAppConditions`  method no modifies app conditions instead of returning new list. So this line is not required any more.